source/Parser_expressions.cpp

```diff
@@ -221,8 +221,17 @@ ExpressionSyntax* Parser::parsePrimaryExpression() {
             // 3. implicit class handles
             // 4. any of [1-3] with an assignment pattern
             // 5. any of [1-3] with a cast expression
-            // 6. error
-            if (isPossibleDataType(kind) && kind != TokenKind::Identifier && kind != TokenKind::UnitSystemName) {
+            // 6. signing cast expression
+            // 7. error
+            if (kind == TokenKind::SignedKeyword || kind == TokenKind::UnsignedKeyword) {
```

Comment:
This looks good, but could you make these case labels above? That avoids the extra branch.